src/js/disability-benefits/526EZ/components/SelectArrayItemsWidget.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import get from '../../../../platform/utilities/data/get';
+import set from '../../../../platform/utilities/data/set';
+
+// TODO: Safety checks for `selected` callback and `label` element
+
+export default class SelectArrayItemsWidget extends React.Component {
+  onChange = (index, checked) => {
+    const items = set(`[${index}].${this.props.options.selectedPropName || this.defaultSelectedPropName}`, checked, this.props.value);
+    this.props.onChange(items);
+  }
+
+  defaultSelectedPropName = 'view:selected'
+
+  render() {
+    const {
+      value: items,
+      id,
+      options,
+      required
+    } = this.props;
+    const { label: Label, selectedPropName, disabled } = options;
+
+    // Note: Much of this was stolen from CheckboxWidget
```

Comment:
Should this be named ErrorableCheckboxesWidget or CheckboxesWidget, similar to the radio widget naming convention?

Reply:
Unlike other fields of the kind, this takes an array of objects and creates "boolean" inputs, so I wanted a name that conveyed that the kind of data it takes is an array of user data.

I'm game to rename, but I don't think `ErrorableCheckboxesWidget` or `CheckboxesWidget` quite fit the bill.

Reply:
Isn't this the default for checkboxes, that a set of checkboxes would use an array of objects?

Reply:
I don't think so. I looked at the `CheckboxWidget` we have already and it just made a single input where `checked={typeof value === 'undefined' ? false : value}`. The new widget takes an array and creates a list of inputs where `checked` is based on the value of the `view:selected` field (or custom option) of the underlying object.

Reply:
Ok, I must've been thinking of the other Checkboxes widget used in the wizard, which seems fairly similar, but perhaps it's not the default use case/configuration for checkboxes.